lib/util.js

```diff
@@ -942,9 +944,19 @@ function formatWeakMap(ctx, value, recurseTimes, keys) {
   return output;
 }
 
+function zip2(list) {
+  const ret = Array(list.length / 2);
+  for (let i = 0; i < ret.length; ++i)
+    ret[i] = [list[2 * i], list[2 * i + 1]];
+  return ret;
+}
```

Comment:
We could use the same pattern as in weak maps instead of first zipping this. This step is just overhead (even though the result is nicer to read).

Reply:
Yes, but it gets pretty tricky to do that, at least for me… we are passing the individual pair arrays to `formatValue`, and it’s not obvious how/whether to short-circuit around that

(Extreme edge case: Calling `formatValue()` on the individual keys/values rather than the pairs even makes a difference in `depth`…)

Reply:
I am not sure in what way this should have any influence on the `depth`?